luluisher/products/views.py

```diff
@@ -0,0 +1,78 @@
+from django.http import JsonResponse
+from django.views import View
+from django.db.models import Q
+
+from products.models import Menu, Product, OptionSize, OptionColor, ProductOption, ProductOptionImage, Activity
+
+class productListView(View):
+    def get(self, request):
+        try: 
+            offset         = int(request.GET.get('offset', 0))
+            limit          = int(request.GET.get('limit', 9))
+            menu           = request.GET.get('menu', None)
+            main_category  = request.GET.get('main_category', None)
+            sub_category   = request.GET.get('sub_category', None)
+            option_color   = request.GET.getlist('option_color', None)
+            option_size    = request.GET.getlist('option_size', None)
+            is_new         = request.GET.get('is_new', None)
+            is_bestseller  = request.GET.get('is_bestseller', None)
+            summer_clothes = request.GET.get('summer_clothes', None)
+            activity       = request.GET.get('activity', None)
+            
+            q = Q()
+            
+            if menu:
+                q &= Q(sub_category__main_category__menu__name=menu)
+                
+            if main_category:
+                q &= Q(sub_category__main_category__name=main_category)
+                
+            if sub_category:
+                q &= Q(sub_category__name=sub_category)
+
+            if option_color:
+                q &= Q(productoption__color__name__in=option_color) 
+                
+            if option_size:
+                q &= Q(productoption__size__name__in=option_size)
+
+            if is_new:
+                q &= Q(is_new__name=is_new)
+                
+            if is_bestseller:
+                q &= Q(is_bestseller__name=is_bestseller)
+
+            if summer_clothes:
+                q &= Q(summer_clothes__name=summer_clothes)
+
+            if activity:
+                q &= Q(activity__activity_name=activity)
+                
+            products = Product.objects.filter(q)[offset:limit]
+
+            if (menu or main_category or sub_category or option_color or option_size or is_new or is_bestseller or summer_clothes or activity) == None:
+                products = Product.objects.all()[offset:limit]
+                
+            product_list = [{
+                'product_id'         : product.id,
+                'name'               : product.name,
+                'original_price'     : product.price,
+                'is_new'             : product.is_new,
+                'is_bestsellers'     : product.is_bestseller,
+                'summer_clothes_shop': product.summer_clothes,
+                'activities'         : [i.activity_name for i in Activity.objects.filter(products__id=product.id)],
+                'product_options': [{
+                    'product_options_id'   : product_options.id,
+                    'color'                : OptionColor.objects.get(id = product_options.color_id).name,
```

Comment:
Line 66, 67, 77에 보면 데이터를 가져올 때 Model.objects를 사용하고 있습니다.

이러한 방식의 코드는 추후에 최적화 할 때 문제가 됩니다. 이 방식이 아니라 이미 생성한 인스턴스의 . 노테이션을 사용해서 참조 관계에 있는 데이터를 가져올 수 있도록 수정해주세요!


```suggestion
                    'color' : product_option.....
```